### PR TITLE
custom git user input option

### LINKED ISSUE #126: Add option to enable commit signing
# Context

At the moment, if you use this action and have the `Require signed commits` setting enabled for a protected branch, the PR generated can't be merged.

# Example

[Crowdin](https://crowdin.com/), a popular localisation management tool, has a similar [action](https://github.com/crowdin/github-action) which opens a PR using a bot.

They allow you to pass a `gpg_private_key` as a config option, which will enable signed commits for the bot opening the PR. See options [here](https://github.com/crowdin/github-action#supported-options).

### PR DESCRIPTION
**Pull request by jacklesliewise**
# Context

Fixes #126, alternative solution to #127

Instead of implementing git commit signing into the action, we could allow an input option to disable the `setupUser` function so that consumers can use their own git user to create commits in this action.

# Changes

- Add `setupGitUser` boolean input option
  - Updated `@actions/core` to be able to use `getBooleanInput`
  - Default to `true`
- Don't run `gitUtils.setupUser()` if `setupGitUser` is `false`
- Update docs

### CODE REVIEW

**Comment by jacklesliewise** [comment]
@Andarist @mitchellhamilton just bumping this if that's okay

Would you prefer this approach or the one in #127?

File: action.yml [hunk 1]
```diff
@@ -17,6 +17,9 @@ inputs:
   title:
     description: The pull request title. Default to `Version Packages`
     required: false
+  customGitUser:
```
Comment (Andarist) [inline action.yml:20]: I would prefer for this to be called something like `setupGitUser` - with the default being `true`.

**Comment by Andarist** [comment]
@jacklesliewise the only concern I have with this PR is the nit about the name that has been chosen for the option. I think this might be useful to some people in other situations than the one covered by #127 so this should land either way (regardless of what we decide for the #127 but most likely we'll close it, for the time being, we can always revisit this in the future).

**Comment by jacklesliewise** [comment]
Thanks @Andarist, I've closed #127 for now and updated this PR as per your comment

File: action.yml (hunk 1 above)
Comment (jacklesliewise) [reply action.yml]: Sounds good, updated

File: .changeset/two-dolls-promise.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@changesets/action": minor
+---
+
+add a `setupGitUser` to enable or disable setting up a default git user
```
Comment (mitchellhamilton) [inline .changeset/two-dolls-promise.md:5]:
```suggestion
Added `setupGitUser` option to enable or disable setting up a default git user
```

**Review by mitchellhamilton — APPROVED** [review]

**Pull request merged**